packages/core/src/stylable-imports-tools.ts

```diff
@@ -0,0 +1,430 @@
+import type { Imported } from './stylable-meta';
+import { Root, decl, Declaration, atRule, rule, Rule, AtRule } from 'postcss';
+import { rootValueMapping, SBTypesParsers, valueMapping } from './stylable-value-parsers';
+import { stripQuotation } from './utils';
+import { isCompRoot } from './helpers/selector';
+
+const parseNamed = SBTypesParsers[valueMapping.named];
+
+export const parseImportMessages = {
+    ST_IMPORT_STAR() {
+        return '@st-import * is not supported';
+    },
+    INVALID_ST_IMPORT_FORMAT(errors: string[]) {
+        return `Invalid @st-import format:\n - ${errors.join('\n - ')}`;
+    },
+
+    ST_IMPORT_EMPTY_FROM() {
+        return '@st-import must specify a valid "from" string value';
+    },
+    EMPTY_IMPORT_FROM() {
+        return '"-st-from" cannot be empty';
+    },
+
+    MULTIPLE_FROM_IN_IMPORT() {
+        return `cannot define multiple "${valueMapping.from}" declarations in a single import`;
+    },
+    DEFAULT_IMPORT_IS_LOWER_CASE() {
+        return 'Default import of a Stylable stylesheet must start with an upper-case letter';
+    },
+    ILLEGAL_PROP_IN_IMPORT(propName: string) {
+        return `"${propName}" css attribute cannot be used inside ${rootValueMapping.import} block`;
+    },
+    FROM_PROP_MISSING_IN_IMPORT() {
+        return `"${valueMapping.from}" is missing in ${rootValueMapping.import} block`;
+    },
+};
+
+export const ensureImportsMessages = {
+    ATTEMPT_OVERRIDE_SYMBOL(
+        kind: 'default' | 'named' | 'keyframes',
+        origin: string,
+        override: string
+    ) {
+        return `Attempt to override existing ${kind} import symbol. ${origin} -> ${override} `;
+    },
+};
+
+export function createAtImportProps(
+    importObj: Partial<Pick<Imported, 'named' | 'keyframes' | 'defaultExport' | 'request'>>
+): {
+    name: string;
+    params: string;
+} {
+    const named = Object.entries(importObj.named || {});
+    const keyframes = Object.entries(importObj.keyframes || {});
+    let params = '';
+    if (importObj.defaultExport) {
+        params += importObj.defaultExport;
+    }
+    if (importObj.defaultExport && (named.length || keyframes.length)) {
+        params += ', ';
+    }
+    if (named.length || keyframes.length) {
+        params += '[';
+
+        const namedParts = getNamedImportParts(named);
+        const keyFramesParts = getNamedImportParts(keyframes);
+
+        params += namedParts.join(', ');
+
+        if (keyFramesParts.length) {
+            if (namedParts.length) {
+                params += ', ';
+            }
+            params += `keyframes(${keyFramesParts.join(', ')})`;
+        }
+        params += ']';
+    }
+
+    params += ` from ${JSON.stringify(importObj.request || '')}`;
+    return { name: 'st-import', params };
+}
+
+export function ensureStylableImports(
+    ast: Root,
+    importPatches: Array<ImportPatch>,
+    options: {
+        mode: 'patch-only' | 'st-import' | ':import';
```

Comment:
I'm not sure about the `mode` option. `patch-only` works on both types - that is the prefered mode, why give any other option? Maybe add a different option `newImportType` to handle addition of non existing imports (with default to `st-import`)